amazonka/src/Network/AWS/Auth.hs

```diff
@@ -514,8 +514,15 @@ fromFilePath profile cred conf =
         else do
           ini <- INI.readIniFile path >>= either (throwInvalid path Nothing) pure
 
-          case INI.lookupValue profile confRegion ini of
-            Left _ -> empty
+          let configProfile =
+                case INI.sections ini of
+                  sections@(_:_)
+                    | profile `elem` sections -> pure profile
+                    | ("profile " <> profile) `elem` sections -> "profile " <> profile
+                  _ -> throwInvalid path Nothing ("profile " <> profile <> " does not exist in config file")
+
+          case INI.lookupValue configProfile confRegion ini of
```

Comment:
Optional suggestion: you might consider achieving this more simply by Alternative instance of Either:
```haskell
case INI.lookupValue configProfile configRegion ini 
  <|> INI.lookupValue ("profile " <> configProfile) configRegion ini of
  ...
```

Reply:
Thanks for the suggestion, but I would rather use [`hush`](https://hackage.haskell.org/package/errors-2.3.0/docs/Control-Error-Util.html#v:hush) than using `<>` for this purpose.

We also lose our warning in case we're missing the necessary profile section with this change, but maybe that's not the problem of `amazonka` anyway.

Reply:
on the other hand, the package doesn't have `errors` library in its dependencies. so hush be available for me anyway. changing it to your suggestion. (still waiting for my nix environment to come up)